Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp

```diff
@@ -129,6 +236,19 @@ VideoFrameGStreamer::VideoFrameGStreamer(const GRefPtr<GstSample>& sample, const
 {
 }
 
+void VideoFrame::paintInContext(GraphicsContext& context, const FloatRect& destination, const ImageOrientation& destinationImageOrientation, bool shouldDiscardAlpha)
+{
+    RELEASE_ASSERT(isGStreamer());
+    auto image = GstSampleColorConverter::singleton().convertSampleToImage(downcast<VideoFrameGStreamer>(*this).sample());
```

Comment:
Since you enabled type-casting, maybe the RELEASE_ASSERT isn't necessary anymore. Using `downcast<>`, there's an assert used inside that, but it's debug-time only.